contrib/win32/win32compat/misc.c

```diff
@@ -237,9 +242,33 @@ dlclose(HMODULE handle)
 FARPROC 
 dlsym(HMODULE handle, const char *symbol)
 {
-	return GetProcAddress(handle, symbol);
+	void *ptr = GetProcAddress(handle, symbol);
+	if (ptr == NULL)
+		last_dlerror = GetLastError();
+	return ptr;
 }
 
+char *
+dlerror()
+{
+	static char *message = NULL;
+	if (message != NULL) {
+		LocalFree(message);
+		message = NULL;
+	}
+	if (last_dlerror == ERROR_SUCCESS)
+		return NULL;
+
+	DWORD length = FormatMessageA(
```

Comment:
Can you please double check if FormatMessageA works for Unicode strings and spits out UTF-8 output? 

If that's not the case, I believe you should be using FormatMessageW followed by utf16_to_utf8.

Reply:
This will presumably use the current code page rather than UTF-8. I will change it to use FormatMessageW.